### PR TITLE
feat: Add cfg-utils crate

### PR DESCRIPTION
**Pull request by NunoAlexandre**
We add this crate to place general utils that can be used across other pallets, runtimes, tests, etc.

To bootstrap it, we added and test a `vec_to_fixed_array` function that is used in https://github.com/centrifuge/centrifuge-chain/pull/1083

### CODE REVIEW

**Review by mustermeiszer — DISMISSED** [review]
LGTM! I Wonder if we should also have a math lib for all the calculations we are doing in pools and loans? 

Cc: @branan @thea-leake @gruberb

File: libs/utils/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+[package]
+authors = ["Centrifuge <admin@centrifuge.io>"]
+description = 'Utility functions for Centrifuge'
+edition = '2021'
+license = "LGPL-3.0"
+name = 'cfg-utils'
+repository = "https://github.com/centrifuge/centrifuge-chain"
```
Comment (lemunozm) [inline libs/utils/Cargo.toml:7]: In some crates of our workspace, the link continues with something like `/libs/utils`, but not sure what should be the correct one

**Review by lemunozm — APPROVED** [review]
Looks perfect to me! 🚀

**Comment by gruberb** [comment]
> LGTM! I Wonder if we should also have a math lib for all the calculations we are doing in pools and loans? 
> 
> Cc: @branan @thea-leake @gruberb 

Yeah, this also came to my mind during my WIP PR about the pool types. Depends if this counts as invariant though!

File: libs/utils/Cargo.toml (hunk 1 above)
Comment (NunoAlexandre) [reply libs/utils/Cargo.toml:7]: Yeah I noticed that when I copy-pasted this cargo file. Those links are all 404 and I don't see the need to make them super specific to the sub-crate, just pointing to the repo itself is fine. I fixed that here: https://github.com/centrifuge/centrifuge-chain/pull/1086

**Comment by mustermeiszer** [comment]
> > LGTM! I Wonder if we should also have a math lib for all the calculations we are doing in pools and loans? 
> > 
> > Cc: @branan @thea-leake @gruberb 
> 
> Yeah, this also came to my mind during my WIP PR about the pool types. Depends if this counts as invariant though!

I would say, basic computations shouldn't be. We also have fixed point arithmetic separate. But rebalancing definitely is System specific. But let's say interest calculation and so on maybe not.

**Pull request merged**